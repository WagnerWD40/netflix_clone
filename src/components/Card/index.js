import { useState, useContext, createContext } from 'react';

import {
    Container,
    Group,
    Title,
    SubTitle,
    Text,
    Meta,
    Image,
    Item,
    Feature,
    FeatureTitle,
    FeatureText,
    FeatureClose,
    Maturity,
    Content,
    Entities,
} from './styles/card';

export const FeatureContext = createContext({});

function Card({ children, ...restProps }) {
    const [showFeature, setShowFeature] = useState(false);
    const [itemFeature, setItemFeature] = useState({});

    const featureContextItems = {
        showFeature,
        setShowFeature,
        itemFeature,
        setItemFeature,
    }

    return (
        <FeatureContext.Provider value={featureContextItems}>
            <Container {...restProps}>{children}</Container>
        </FeatureContext.Provider>
    );
}

Card.Group = function ({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>;
}

Card.Title = function ({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>;
}

Card.SubTitle = function ({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>;
}

Card.Text = function ({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
}

Card.Meta = function ({ children, ...restProps }) {
    return <Meta {...restProps}>{children}</Meta>;
}

Card.Feature = function CardFeature({ category, children, ...restProps }) {


    const { showFeature, itemFeature, setShowFeature } = useContext(FeatureContext);

    function handleClose() {
        setShowFeature(false);
    }

    return showFeature && ( 
                <Feature 
                    {...restProps}
                    src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}>
                    <Content>
                        <FeatureTitle>{itemFeature.title}</FeatureTitle>
                        <FeatureText>{itemFeature.description}</FeatureText>
                        <FeatureClose onClick={handleClose}>
                            <img src="/images/icons/close.png" alt="Close" />
                        </FeatureClose>
                    
                        <Group margin="30px 0" flexDirection="row" alignItems="center">
                            <Maturity rating={itemFeature.maturity}>
                                {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
                            </Maturity>
                            <FeatureText fontWeight="bold">
                                {itemFeature.genre}
                            </FeatureText>
                        </Group>
                        {children}
                    </Content>
                </Feature>
            );
}

Card.Item = function CardItem({ item, children, ...restProps }) {
    const { setShowFeature, setItemFeature } = useContext(FeatureContext);

    function handleClick() {
        setItemFeature(item);
        setShowFeature(true);
    }

    return <Item 
                onClick={handleClick}
                {...restProps}>{children}</Item>;
}

Card.Image = function ({ ...restProps }) {
    return <Image {...restProps} />
}

Card.Entities = function ({ ...restProps }) {
    return <Entities {...restProps} ></Entities>
}

export default Card;