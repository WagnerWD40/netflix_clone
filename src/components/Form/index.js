import {
    Container,
    Base,
    Error,
    Title,
    Text,
    TextSmall,
    Link,
    Input,
    Submit,
} from './styles/form';

function Form({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Form.Base = function ({ children, ...restProps }) {
    return <Base {...restProps}>{children}</Base>
}

Form.Error = function ({ children, ...restProps }) {
    return <Error {...restProps}>{children}</Error>
}

Form.Title = function ({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Form.Text = function ({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Form.TextSmall = function ({ children, ...restProps }) {
    return <TextSmall {...restProps}>{children}</TextSmall>
}

Form.Link = function ({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>
}

Form.Input = function ({ children, ...restProps }) {
    return <Input {...restProps}>{children}</Input>
}

Form.Submit = function ({ children, ...restProps }) {
    return <Submit {...restProps}>{children}</Submit>
}


export default Form;