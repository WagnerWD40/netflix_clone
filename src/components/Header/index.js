import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';

import {
    Background,
    Container,
    Group,
    Search,
    SearchIcon,
    SearchInput,
    Feature,
    FeatureCallOut,
    Text,
    ButtonLink,
    PlayButton,
    Link,
    Logo,
    Profile,
    Dropdown,
    Picture,
} from './styles/header';

function Header({ bg = true, children, ...restProps }) {
return bg ? <Background {...restProps} data-testid="header-bg">{children}</Background> : children;
}

Header.Feature = function ({ children, ...restProps }) {
    return <Feature {...restProps}>{children}</Feature>;
}

Header.FeatureCallOut = function ({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [searchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps}>
            <SearchIcon onClick={() => setSearchActive(searchActive => !searchActive)}>
                <img src="/images/icons/search.png" alt="search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={searchActive} />
        </Search>
    )
}

Header.Frame = function ({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
}

Header.Group = function ({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>;
}

Header.Text = function ({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Header.Dropdown = function ({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>;
}

Header.Picture = function ({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />;
}

Header.Profile = function ({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>;
}

Header.ButtonLink = function ({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>;
}

Header.TextLink = function ({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>;
}

Header.PlayButton = function({ children, ...restProps }) {
    return <PlayButton {...restProps}>{children}</PlayButton>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to} {...restProps}>
            <Logo {...restProps} />
        </ReactRouterLink>
    );
}

export default Header;