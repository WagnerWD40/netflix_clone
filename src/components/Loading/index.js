import React from 'react';

import {
    Spinner,
    LockBody,
    ReleaseBody,
    Picture,
} from './styles/loading';

function Loading({ src, ...restProps }) {
    return (
        <Spinner>
            <LockBody />
            <Picture src={`/images/users/${src}.png`} />
        </Spinner>
    );
}

Loading.ReleaseBody = function () {
    return <ReleaseBody />
}

export default Loading;