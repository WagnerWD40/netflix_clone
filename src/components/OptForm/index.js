import {
    Container,
    Input,
    Button,
    Text,
    Break,
} from './styles/optForm';

function OptForm({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

OptForm.Input = function ({...restProps }) {
    return <Input {...restProps} />;
}

OptForm.Button = function ({ children, ...restProps }) {
    return (
        <Button {...restProps}>
            {children} <img src="/images/icons/chevron-right.png" alt="Try Now" />
        </Button>
    );
}

OptForm.Text = function ({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

OptForm.Break = function ({ children, ...restProps }) {
    return <Break {...restProps}>{children}</Break>
}

export default OptForm;