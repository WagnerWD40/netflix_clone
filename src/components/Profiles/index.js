
import {
    Container,
    Title,
    List,
    Item,
    Picture,
    Name,
} from './styles/profiles';

function Profiles({ user, children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
}

Profiles.Title = function ({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>;
}

Profiles.List = function ({ children, ...restProps }) {
    return <List {...restProps}>{children}</List>;
}

Profiles.User = function ({ children, ...restProps }) {
    return <Item {...restProps}>{children}</Item>;
}

Profiles.Picture = function ({ avatar, loading, avatarPic, setAvatarPic, setLoading, ...restProps }) {
    return <Picture
                {...restProps}
                avatar={avatar}
                loading={loading}
                onAnimationEnd={() => avatar ? setAvatarPic(`/images/users/${avatar}.png`) : setLoading(true)}
                src={avatarPic} />;
}

Profiles.Name = function ({ children, ...restProps }) {
    return <Name {...restProps}>{children}</Name>;
}

export default Profiles;