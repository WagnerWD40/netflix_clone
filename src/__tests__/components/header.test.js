import { render, fireEvent } from '@testing-library/react';
import { Header } from '../../components';

// jest.mock('../../components', () => {
//     return mockComponent("Header");
// });
jest.mock('react-router-dom');

describe('<Header />', () => {
    it('renders the basic <Header /> with a background ', () => {
        const { container, getAllByText, getByTestId } = render (
            <Header>
                <Header.Frame>
                    {/* <Header.Logo alt="Netflix" src="/images/logo.svg" /> */}
                    {/* <Header.ButtonLink >Sign In</Header.ButtonLink>
                    <Header.TextLink>Sign In</Header.TextLink> */}
                </Header.Frame>
            </Header>            
        );

        // const [button, textLink ] = getAllByText('Sign In');

        // expect(button).toBeTruthy();
        // expect(textLink).toBeTruthy();
        expect(getByTestId('header-bg')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});