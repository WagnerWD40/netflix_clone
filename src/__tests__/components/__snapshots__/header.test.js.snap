// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background  1`] = `
<div
  class="header__Background-bd9uij-0 FaKIH"
  data-testid="header-bg"
>
  <div
    class="header__Container-bd9uij-6 koFwzB"
  />
</div>
`;
